src/logic/notifications/notificationBuilder.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+import { NOTIFIED_TRANSACTIONS } from '~/logic/safe/transactions'
+import { type Notification, NOTIFICATIONS } from './notificationTypes'
+
+type NotificationsQueue = {
+  beforeExecution: Notification,
+  pendingExecution: {
+    single: Notification,
+    multiple: Notification,
+  },
+  afterExecution: Notification,
+  afterExecutionError: Notification,
+  afterRejection: Notification,
+}
+
+const standardTxNotificationsQueue: NotificationsQueue = {
+  beforeExecution: NOTIFICATIONS.SIGN_TX_MSG,
+  pendingExecution: {
+    single: NOTIFICATIONS.TX_PENDING_MSG,
+    multiple: NOTIFICATIONS.TX_PENDING_MORE_CONFIRMATIONS_MSG,
+  },
+  afterRejection: NOTIFICATIONS.TX_REJECTED_MSG,
+  afterExecution: NOTIFICATIONS.TX_EXECUTED_MSG,
+  afterExecutionError: NOTIFICATIONS.TX_FAILED_MSG,
+}
+
+const confirmationTxNotificationsQueue: NotificationsQueue = {
+  beforeExecution: NOTIFICATIONS.SIGN_TX_MSG,
+  pendingExecution: {
+    single: NOTIFICATIONS.TX_CONFIRMATION_PENDING_MSG,
+    multiple: null,
```

Comment:
I'd rename `multiple` one to something like `pendingExecutionNeedsConfirmations` or `pendingExecution.needsMoreConf`